### PR TITLE
#P6PsRgMK CRUD Operations for Buses

### PR DESCRIPTION
**Pull request by ncutixavier**
### **What about this PR?**
This PR has CRUD Operations for buses

### **Description of Task to be completed**
It should have the following endpoint:
- GET /api/v1/buses

- GET /api/v1/buses?page={page}&limit={limit}

- POST /api/v1/buses

- GET /api/v1/buses/id

- PATCH /api/v1/buses/id

- DELETE /api/v1/buses/id

### **How should be manually tested?**
- Clone the repo
- checkout the branch `ft-roles-and-permissions-zqZkd9M8`
- Install all packages
- Set up dotenv with `.env.example` as reference
- Run this command `npm run migrateDB` to make migration
- You should use the credentials below or you may create your user to access some endpoint that need to be logged in
{
    "email":"admin@phantom.com",
    "password":"admin"
}
- Copy token and paste it in documentation or in header if you are using postman.
- You may also test with the [documentation](https://phantom-caba-ft-crud-fo-pasycn.herokuapp.com/api-docs/)

### **Relevant pivots tracker**
[#P6PsRgMK](https://trello.com/c/P6PsRgMK/11-crud-operations-for-buses)

### CODE REVIEW

**Review by Niyiragira — APPROVED** [review]

**Review by karigirwa — APPROVED** [review]

**Review by Nkbtemmy — APPROVED** [review]

**Review by blaise82 — COMMENTED** [review]
Make sure that you add values to your swagger
and if you don't know why your using bearer in front of a token remove it

**Review by blaise82 — CHANGES_REQUESTED** [review]
adding bus
- user should not be able to add a bus with zero seats

find bus by id
- make sure that the route is projected too

update bus
- make sure that on update the buses can't have the same plate number after the update

File: src/controller/busControllers.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return res.status(500).json({
+                message: res.__("can't add new bus")
+            })
+        }
+    }
+
+    static async getAllBus(req, res) {
+        try {
+            const page = req.query.page || 1
+            const limit = req.query.limit || 10
```
Comment (blaise82) [inline src/controller/busControllers.js]: Use destructuring assignment

File: src/routes/busRoutes.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        busValidation.busValidate,
+        checkExist.checkPlate,
+        busController.createNewBus
+    )
+
+router
+    .route('/:id')
+    .get(
+        checkExist.checkID,
+        busController.getBus
```
Comment (blaise82) [inline src/routes/busRoutes.js]: put your swagger documentation always on the top of a route
not in the swagger.json file

**Review by blaise82 — CHANGES_REQUESTED** [review]

File: src/routes/busRoutes.js (hunk 2 above)
Comment (ncutixavier) [reply src/routes/busRoutes.js]: Done

File: src/controller/busControllers.js (hunk 1 above)
Comment (ncutixavier) [reply src/controller/busControllers.js]: Done

**Review by blaise82 — APPROVED** [review]

**Pull request merged**